openapi/media_type_test.go

```diff
@@ -5,57 +5,47 @@ import (
 	"testing"
 )
 
-func getMediaType() *MediaType {
-	openapi, _ := LoadOpenAPI(openapiFile)
-	mt := openapi.getChildren()["/endpoint"].(*PathItem).getChildren()["post"].(*Operation).getChildren()["RequestBody"].(*RequestBody).getChildren()["application/json"]
-	cast, ok := mt.(*MediaType)
-	if ok {
-		return cast
-	}
-	return nil
-}
-
 func TestMediaType_GetName(t *testing.T) {
-	//mt := getMediaType()
-	t.Errorf("Implement me")
-	t.Fail()
+	openapi, _ := LoadOpenAPI(absOpenAPI)
+	SetRenderer(openapi, GinRenderer{})
+	mt := Dig(openapi, testEndpoint, testMethod, testReqBody, testMediaType)
+	name := mt.GetName()
+	if name != testGinRenderedMediaItemName {
+		t.Errorf("expected %v, got %v", testGinRenderedMediaItemName, name)
+	}
 }
 
 func TestMediaType_GetChildren(t *testing.T) {
-	mt := getMediaType()
-	transversable := mt.getChildren()
-	s, ok := transversable["Model"].(*Schema)
-	if !ok {
-		t.Logf("MediaType.getChildren() didn't return a *Schema")
-		t.Fail()
-	}
-	if s == nil {
-		t.Logf("MediaType.getChildren() returned a nil schema")
-		t.Fail()
+	openapi, _ := LoadOpenAPI(absOpenAPI)
+	mt := Dig(openapi, testEndpoint, testMethod, testReqBody, testMediaType)
+	children := mt.getChildren()
+	for model, schema := range children {
+		if model == testSchema {
+			s, ok := schema.(*Schema)
+			if !ok {
+				t.Errorf("MediaType.getChildren() didn't return a *Schema")
```

Comment:
nonblocking: These tests would be more valuable if we asserted against the content of the schema object we expect to make sure we get the right one. I'll consider sprucing it up at some point. This is already progress

Reply:
TODO'ed it inline